### PR TITLE
chore(ts): migrate behaviors

### PR DESCRIPTION
**Pull request by flochtililoch**

### CODE REVIEW

File: src/behaviors/hv-hide/index.ts [hunk 1]
```diff
@@ -20,7 +16,9 @@ export default {
     getRoot: HvGetRoot,
     updateRoot: HvUpdateRoot,
   ) => {
-    const targetId: ?DOMString = element.getAttribute('target');
+    const targetId: DOMString | null | undefined = element.getAttribute(
```
Comment (hgray-instawork) [inline src/behaviors/hv-hide/index.ts:19]: Did you decide to leave DOMString in the types?

File: src/behaviors/index.ts [hunk 2]
```diff
@@ -33,14 +31,15 @@ const HYPERVIEW_BEHAVIORS = [
   HvUnselectAll,
 ];
 
-export const getRegistry = (behaviors: HvBehavior[] = []): BehaviorRegistry =>
-  [...HYPERVIEW_BEHAVIORS, ...behaviors].reduce(
+export const getRegistry = (behaviors: HvBehavior[] = []): BehaviorRegistry => {
```
Comment (hgray-instawork) [inline src/behaviors/index.ts:34]: Necessary?

**Review by hgray-instawork — APPROVED** [review]

File: src/behaviors/hv-hide/index.ts (hunk 1 above)
Comment (flochtililoch) [reply src/behaviors/hv-hide/index.ts:19]: yeah I wanted to decouple the types renaming to another future PRs, to limit the size of the diff

File: src/behaviors/index.ts (hunk 2 above)
Comment (flochtililoch) [reply src/behaviors/index.ts:34]: I need to investigate what eslint rule causes this, it's happened in most files. I'll update before merging.

Comment (hgray-instawork) [reply src/behaviors/index.ts:34]: I ran into it a bunch in my work. I think it was a conflict between eslint and prettier which didn't allow single line arrow statements.

File: src/behaviors/hv-alert/index.ts [hunk 3]
```diff
@@ -11,27 +9,31 @@
...
+    const childNodes = element.childNodes
+      ? Array.from(element.childNodes as NodeListOf<Element>)
+      : [];
 
     // Get the immediate alert:option nodes. We don't use getElementsByTagname to
     // avoid getting options for nested alerts.
-    const optionElements = childNodes.filter(
-      n =>
+    const optionElements = childNodes.filter(n => {
+      return (
```
Comment (flochtililoch) [inline src/behaviors/hv-alert/index.ts:29]: todo: revert

File: src/behaviors/hv-alert/index.ts [hunk 4]
```diff
@@ -42,9 +44,12 @@ export default {
             // Only behaviors with "press" trigger will get executed.
             // "press" is also the default trigger, so if no trigger is specified,
             // the behavior will also execute.
-            e =>
-              !e.getAttribute('trigger') ||
-              e.getAttribute('trigger') === 'press',
+            e => {
+              return (
```
Comment (flochtililoch) [inline src/behaviors/hv-alert/index.ts:48]: todo: revert

File: src/behaviors/hv-alert/index.ts [hunk 5]
```diff
@@ -62,8 +67,8 @@ export default {
             // With multiple behaviors for the same trigger, we need to stagger
             // the updates a bit so that each update operates on the latest DOM.
             // Ideally, we could apply multiple DOM updates at a time.
-            later(i).then(
-              () =>
+            later(i).then(() => {
+              return (
```
Comment (flochtililoch) [inline src/behaviors/hv-alert/index.ts:71]: todo: revert

File: src/behaviors/index.ts (hunk 2 above)
Comment (flochtililoch) [reply src/behaviors/index.ts:34]: todo: revert

File: src/behaviors/hv-alert/index.ts (hunk 4 above)
Comment (flochtililoch) [reply src/behaviors/hv-alert/index.ts]: a93387d

File: src/behaviors/hv-alert/index.ts (hunk 3 above)
Comment (flochtililoch) [reply src/behaviors/hv-alert/index.ts]: a93387d

File: src/behaviors/index.ts (hunk 2 above)
Comment (flochtililoch) [reply src/behaviors/index.ts]: a93387d

**Pull request merged**